src/cascadia/TerminalApp/TerminalPage.h

```diff
@@ -6,12 +6,144 @@
 #include "winrt/Microsoft.UI.Xaml.Controls.h"
 
 #include "TerminalPage.g.h"
+#include "Tab.h"
+#include "CascadiaSettings.h"
+#include "Profile.h"
+#include "ScopedResourceLoader.h"
+
+#include <winrt/Microsoft.Terminal.TerminalControl.h>
+#include <winrt/Microsoft.Terminal.TerminalConnection.h>
+#include <winrt/Microsoft.UI.Xaml.Controls.Primitives.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
+#include <winrt/Microsoft.UI.Xaml.XamlTypeInfo.h>
 
 namespace winrt::TerminalApp::implementation
 {
     struct TerminalPage : TerminalPageT<TerminalPage>
     {
+    public:
         TerminalPage();
+
+        void SetSettings(std::shared_ptr<::TerminalApp::CascadiaSettings> settings, bool needRefreshUI);
+
+        void SetResourceLoader(std::shared_ptr<ScopedResourceLoader> resourceLoader);
```

Comment:
This seems like it could have come in through the constructor...

Reply:
is there a reason it cannot?

Reply:
We have a TerminalPage() in the TerminalPage.idl, and ScopedResourceLoader is not a winrt type so we can not access it in the idl file. Thus, I think we need a constructor without any input arguments. I changed to use overload constructor.